stack/dispatcher.go

```diff
@@ -0,0 +1,104 @@
+package stack
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/tendermint/tmlibs/log"
+
+	"github.com/tendermint/basecoin"
+	"github.com/tendermint/basecoin/errors"
+	"github.com/tendermint/basecoin/types"
+)
+
+const (
+	NameDispatcher = "disp"
+)
+
+// Dispatcher grabs a bunch of Dispatchables and groups them into one Handler.
+//
+// It will route tx to the proper locations and also allows them to call each
+// other synchronously through the same tx methods.
+type Dispatcher struct {
+	routes map[string]Dispatchable
```

Comment:
non-deterministic ordering if you iterate through them. good call.

here I just do a lookup.  which is deterministic. 

it would be good to add a comment to be careful though.